### PR TITLE
Rename meilisearch-http

### LINKED ISSUE #3073: Rename `meilisearch-http`
Once the pre-release is over, and the change has been ported back into the `main` branch, we should rename the `meilisearch-http` module into `meilisearch`.

### PR DESCRIPTION
**Pull request by colbsmcdolbs**
# Pull Request

## Related issue
Fixes #3073 

## What does this PR do?
- Renames all references of `meilisearch-http` to `meilisearch`
- Might need to be rebased before the 1.0.0 release

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: meilisearch/Cargo.toml
Comment (Kerollmops) [inline meilisearch/Cargo.toml:9]: Could you please remove this `[[bin]]` section, please? 😉

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for this contribution!

**Comment by irevoire** [comment]
Hey, I would prefer to wait until we merge `release-v0.30` into `main` before renaming anything.

**Comment by curquiza** [comment]
I put it as a draft then to avoid any problem

File: meilisearch/Cargo.toml
Comment (colbsmcdolbs) [reply meilisearch/Cargo.toml:9]: @Kerollmops 
`[[bin]]
name = "meilisearch"
path = "src/main.rs"`

This entire section needs to be removed?

Comment (irevoire) [reply meilisearch/Cargo.toml:9]: Yes! You should be able to remove everything because, by default, cargo is going to name the binary after the name of the crate (meilisearch here).

**Comment by curquiza** [comment]
Now all the changes in v0.30.0 are integrated into `main`, you can rebase @colbsmcdolbs so that we can review and merge your PR 😄

**Comment by colbsmcdolbs** [comment]
@curquiza Merge conflicts seem resolved. I believe this is ready for review.

**Comment by curquiza** [comment]
bors try

**Review by Kerollmops — DISMISSED** [review]
According to what I tried on my machine, everything looks good!
Thank you very much for your time and help ❤️

**Review by irevoire — DISMISSED** [review]

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
Hello everyone here, looks like there is a test failing regarding the format 😢

**Comment by curquiza** [comment]
Looks like it's not due to this PR, and [this PR](https://github.com/meilisearch/meilisearch/pull/3215) should fix the issue

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Comment by Kerollmops** [comment]
bors merge

**Pull request merged**
